src/App.jsx

```diff
@@ -14,45 +19,81 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+export const App = () => {
+  const [sortBy, setSortBy] = useState(NO_SORT);
+  const [reversed, setReversed] = useState(false);
+  const [goods, setGoods] = useState([...goodsFromServer]);
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button is-info ${sortBy !== ALPHABET && 'is-light'}`}
+          onClick={() => {
+            sortGoods(
+              ALPHABET,
+              setSortBy,
+              setReversed,
+              setGoods,
+              [...goodsFromServer],
+              reversed,
+            );
+          }}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={`button is-success ${sortBy !== LENTGH && 'is-light'}`}
+          onClick={() => {
+            sortGoods(
+              LENTGH,
+              setSortBy,
+              setReversed,
+              setGoods,
+              [...goodsFromServer],
+              reversed,
+            );
+          }}
+        >
+          Sort by length
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          type="button"
+          className={`button is-warning ${reversed !== true && 'is-light'}`}
+          onClick={() => {
+            reverseGoods(goods, reversed, setReversed, setGoods);
+          }}
+        >
+          Reverse
+        </button>
+
+        {(sortBy !== NO_SORT || reversed === true) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              sortGoods(
+                NO_SORT,
+                setSortBy,
+                setReversed,
+                setGoods,
+                [...goodsFromServer],
+                reversed,
+              );
+            }}
+          >
+            Reset
+          </button>
+        )}
+      </div>
+
+      <ul>
+        {goods.map(good => <li data-cy="Good">{good}</li>)}
```

Comment:
Each child in a list should have a unique 'key' prop. Add a unique 'key' prop to the <li> component.

Reply:
but how I can do it? Goods doesn't have any id or smth etc. The names of goods may repeat. If I use index as key, I get error that I should'n use index as key